src/modal/bs-modal.service.ts

```diff
@@ -134,7 +134,7 @@ export class BsModalService {
       .provide({ provide: BsModalRef, useValue: bsModalRef })
       .attach(ModalContainerComponent)
       .to('body');
-    bsModalRef.hide = () => modalContainerRef.instance?.hide();
+    bsModalRef.hide = () => this.hide(bsModalRef.id);
```

Comment:
This is how the modal is hide when calling from the service itself instead of from the reference. The action is immediate.

Reply:
Sadly This change broke the close interceptor 😥
https://github.com/valor-software/ngx-bootstrap/issues/6608